src/test/timedata_tests.cpp

```diff
@@ -34,4 +34,19 @@ BOOST_AUTO_TEST_CASE(util_MedianFilter)
     BOOST_CHECK_EQUAL(filter.median(), 7);
 }
 
+BOOST_AUTO_TEST_CASE(util_MedianFilterShallNotGrowBeyondSize)
+{
+    CMedianFilter<int> filter(2, 15);
+
+    BOOST_CHECK_EQUAL(filter.size(), 1); // 15
+
+    filter.input(100); // 15 100
+    BOOST_CHECK_EQUAL(filter.size(), 2);
+
+    filter.input(10); // 100 10
+    BOOST_CHECK_EQUAL(filter.size(), 2);
```

Comment:
Maybe add an unsorted filter check like this?

```
    BOOST_CHECK_EQUAL(filter[0], 100);
    BOOST_CHECK_EQUAL(filter[1], 10);
```

Reply:
Hi Jimmy. Class under test is not a vector, so index operations are not available. Otherwise good suggestion. Code is from 2014. Am focusing on tests in order to nail the contract as a basis for potential improvements in the future.